ckb/ckb-next: 0.2.9 -> 0.3.2, and cleanup

kierdavis opened the pull request:
* updated ckb package from 0.2.9 to 0.3.0 (EDIT: now 0.3.2)
* renamed package and module from "ckb" to "ckb-next", following the move by the upstream developers to do the same
* minor tidy-ups

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

kierdavis committed:
ckb-next: 0.2.9 -> 0.3.2

In this update:

* binaries `ckb` and `ckb-daemon` are renamed to `ckb-next` and `ckb-next-daemon`
* build system changed from qmake to cmake
* the directory searched for animation plugins no longer needs to be patched, as a result of the build system change
* modprobe patch has been bumped, since the source repository layout has changed
* the cmake scripts are quite FHS-centric and require patching to fix install locations

kierdavis committed:
ckb module: update systemd service parameters to match upstream

This changes the description and restart mode to the values present
in lib/systemd/system/ckb.service within the ckb package.

kierdavis committed:
ckb, ckb module: rename to ckb-next

The upstream package has officially changed its name to ckb-next.

kierdavis committed:
ckb module: use exec when starting the daemon process

This avoids leaving the parent shell process (the one executing the
unit script) lying around.

kierdavis committed:
ckb-next: rename ckb-modprobe.patch to modprobe.patch

This keeps the naming consistent with the other two patches in
this directory.

samueldr commented on pkgs/top-level/all-packages.nix:
```diff
@@ -1768,7 +1768,8 @@ with pkgs;
 
-  ckb = libsForQt5.callPackage ../tools/misc/ckb { };
+  ckb-next = libsForQt5.callPackage ../tools/misc/ckb-next { };
+  ckb = ckb-next; # alias retained for backwards compatibility
```
Please move the alias to `pkgs/top-level/aliases.nix`.

(A bunch of aliases are marked as `TODO: move to aliases.nix` in this file.)

samueldr commented on pkgs/tools/misc/ckb-next/default.nix:
```diff
@@ -1,14 +1,14 @@
 
 stdenv.mkDerivation rec {
-  version = "0.2.9";
+  version = "0.3.0";
```
`0.3.1` was released meanwhile, if it isn't too much, update and verify it still works (assuming you have the hardware.)

samueldr reviewed (changes requested):
Thanks!

Minor changes requested, see the comments. Other than those nitpicks it looks like an approval from me.

I would additionally like if you could add the option rename to the release-notes (`nixos/doc/manual/release-notes/rl-1809.xml`). If you can't or don't want to write the release notes just ask and I'll handle it.

kierdavis commented:
@samueldr I've pushed an updated version of this PR. Changes are:

* Upgraded to version 0.3.2
* Rebased onto master to fix merge conflicts
* Moved alias to `pkgs/top-level/aliases.nix` as requested
* Added an entry to the 19.03 release notes

P.S. Github's UI appears to be behaving weirdly for me: on some page loads, the head commit of `kierdavis/ckb-update-and-cleanup` is wrong, resulted in outdated logs/diffs being shown as if they are current. When reviewing please double-check that the most recent commit on the branch is `7a0d8496537a8b31a0697a8d08fa042837fb467c`.

samueldr commented on nixos/modules/hardware/ckb-next.nix:
```diff
@@ -3,17 +3,22 @@
+    imports = [
+      (mkRenamedOptionModule ["hardware" "ckb" "enable"] ["hardware" "ckb-next" "enable"])
+      (mkRenamedOptionModule ["hardware" "ckb" "package"] ["hardware" "ckb-next" "package"])
+    ];
```
Oops, looks like I missed those renames. I personally don't mind if those are kept local to the relevant module, but it looks like most renames are in `nixos/modules/rename.nix`.

I wouldn't hold back the change on account of that, though let's say what you yourself think about it, and if others have a strong opinion about that.

samueldr reviewed (approved):
Other than the really nit-picky comment, and the merge conflict, this looks fine. Everything fixed as asked.

kierdavis committed:
Merge branch 'master' into ckb-update-and-cleanup

kierdavis committed:
ckb-next: move option renames to nixos/modules/rename.nix for consistency

kierdavis commented:
@samueldr Thanks for the review! I've fixed the merge conflict and moved the option renames as suggested.

kierdavis commented:
Thanks!
